src/System.Management.Automation/engine/Modules/PSModuleInfo.cs

```diff
@@ -834,6 +834,19 @@ internal void AddToCompatiblePSEditions(string psEdition)
             _compatiblePSEditions.Add(psEdition);
         }
 
+        internal void AddToCompatiblePSEditions(IEnumerable<string> psEditions)
+        {
+            _compatiblePSEditions.AddRange(psEditions);
+        }
+
+        /// <summary>
+        /// Describes whether the module was considered compatible at load time.
+        /// Any module not on the System32 module path should have this as true.
+        /// Modules loaded from the System32 module path will have this as true if they
+        /// have declared edition compatibility with PowerShell Core.
+        /// </summary>
+        internal bool IsConsideredEditionCompatible { get; set; } = true;
```

Comment:
With this default value, directly importing a non-psd1 module from the system32 module path will succeed.
This might not be the desired behavior, but let's defer solving that problem.

Reply:
Please open an issue tracking this.